src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationProvider.cs

```diff
@@ -197,16 +201,39 @@ private async Task LoadAsync()
             {
                 foreach (var secretDescriptor in secretDescriptors)
                 {
-                    var result = await client.GetSecretAsync(store, secretDescriptor.SecretName, secretDescriptor.Metadata).ConfigureAwait(false);
+                    
+                    Dictionary<string, string> result;
+
+                    try
+                    {
+                        result = await client
+                            .GetSecretAsync(store, secretDescriptor.SecretKey, secretDescriptor.Metadata)
+                            .ConfigureAwait(false);
+                    }
+                    catch (DaprException e)
+                    {
+                        if (secretDescriptor.IsRequired)
+                        {
+                            throw e;
+                        }
+                            
+                        Console.WriteLine($"'{secretDescriptor.SecretName}' doesn't exists in Key Vault but because " +
+                                          $"it doesn't require existence so all errors suppressed!");
```

Comment:
I would remove this unless there is a logger available. Then you could make it a debug statement.

Reply:
Fixed